### ISSUE TITLE
False positive for Style/HashSlice with nested arrays

### ISSUE THREAD

**Issue by KevinCarterDev** [rubocop/rubocop#13756]
The new Style/HashSlice cop (https://github.com/rubocop/rubocop/pull/13507) incorrectly suggests replacing `select` with `slice` on arrays of arrays.

--------

## Expected behavior

No offense should be reported for array of arrays.

## Actual behavior

An incorrect offense is reported on the array of arrays

## Steps to reproduce the problem

With the following contents in a file named `test.rb`
```ruby
# frozen_string_literal: true

array = [
  [:foo, 1],
  [:bar, 2],
  [:baz, 3]
]

keep = %i[foo baz]

final = array.select { |name, _value| keep.include?(name) }

puts final.inspect
```

```bash
% ruby test.rb
[[:foo, 1], [:baz, 3]]

% rubocop test.rb
Inspecting 1 file
C

Offenses:

test.rb:11:15: C: [Correctable] Style/HashSlice: Use slice(*keep) instead.
final = array.select { |name, _value| keep.include?(name) }
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

1 file inspected, 1 offense detected, 1 offense autocorrectable
```

Applying the "correction"
```ruby
# frozen_string_literal: true

array = [
  [:foo, 1],
  [:bar, 2],
  [:baz, 3]
]

keep = %i[foo baz]

final = array.slice(*keep)

puts final.inspect
```

```bash
% ruby test.rb
test.rb:11:in `slice': no implicit conversion of Symbol into Integer (TypeError)

final = array.slice(*keep)
                    ^^^^^
        from test.rb:11:in `<main>'
```

## RuboCop version

Include the output of `rubocop -V` or `bundle exec rubocop -V` if using Bundler.
If you see extension cop versions (e.g. `rubocop-performance`, `rubocop-rspec`, and others)
output by `rubocop -V`, include them as well. Here's an example:

```
$ rubocop -V
1.71.0 (using Parser 3.3.7.0, rubocop-ast 1.37.0, analyzing as Ruby 3.3, running on ruby 3.3.6) [arm64-darwin24]
```

**Comment by koic** [comment]
RuboCop isn't possible to know the receiver's object type or data structure, so this cop is already marked as unsafe. Thank you for the feedback.
https://docs.rubocop.org/rubocop/1.71/cops_style.html#stylehashslice
